include/openPMD/IO/ADIOS/ADIOS2IOHandler.hpp

```diff
@@ -509,15 +509,34 @@ class ADIOS2IOHandlerImpl
         }
         else
         {
-            for (unsigned int i = 0; i < actualDim; i++)
+            if (var.ShapeID() == adios2::ShapeID::JoinedArray)
             {
-                if (!(joinedDim.has_value() && *joinedDim == i) &&
-                    offset[i] + extent[i] > shape[i])
+                // When you reach here due to a mysterious inconsistency
+                // observed from time to time e.g.
+                // adios2::JoinedDim=(~(size_t)0) which for 64 bits =
+                // 18446744073709551615 but the particle shape,  when printed
+                // out,  turns out to be [18446744073709551614]
+                //
+                if (!offset.empty())
                 {
                     throw std::runtime_error(
-                        "[ADIOS2] Dataset access out of bounds.");
+                        "[ADIOS2] Offset must be an empty vector in case of "
+                        "joined "
+                        "array.");
```

Comment:
This duplicates the condition checked in the if branch from line 492 to 509. 
Can't we just change the if condition in line 492 to:
```cpp
if (joinedDim.has_value() || var.ShapeID() == adios2::ShapeID::JoinedArray)
```
This would be a single-line change and do basically the same

Reply:
That issue only comes up when `joinedDim.has_value()` is false, but `var.ShapeID() == adios2::ShapeID::JoinedArray` is true, which we should treat as an error anyway?
I think I'll add a check for that circumstance.